src/zns-route-connect.tsx

```diff
@@ -25,20 +25,23 @@ export class Container extends React.Component<Properties> {
   }
 
   componentDidMount() {
-    this.props.setRoute(this.extractRouteFromProps());
+    const routeApp: RouteApp = { route: this.extractRouteFromProps(), hasAppChanged: false };
+    this.props.setRoute(routeApp);
     this.props.setSelectedApp(this.extractAppFromProps());
   }
 
   componentDidUpdate(prevProps: Properties) {
     const currentRoute = this.extractRouteFromProps();
     const selectedApp = this.extractAppFromProps();
-
-    if (currentRoute !== this.extractRouteFromProps(prevProps)) {
-      this.props.setRoute(currentRoute);
-    }
+    let routeApp: RouteApp = { route: currentRoute, hasAppChanged: false };
 
     if (selectedApp !== this.extractAppFromProps(prevProps)) {
       this.props.setSelectedApp(selectedApp);
+      routeApp.hasAppChanged = true;
+    }
+
+    if (currentRoute !== this.extractRouteFromProps(prevProps)) {
+      this.props.setRoute(routeApp);
     }
```

Comment:
i *think* we probably still want this to be before `setSelectedApp`. this will work totally fine as it is, but my reasoning is that setting the route first will make it less likely that the app will load with the old route, and then have to immediately update. it's possible that having it this way will lead to more requests, or a flicker, etc. depending on how the app is set up.